k8s/persistent_volumes_test.go

```diff
@@ -143,3 +145,140 @@ func TestUploadToVolume(t *testing.T) {
 		t.Error("unexpected output")
 	}
 }
+
+func TestServeRsyncOnVolume(t *testing.T) {
+	var err error
+
+	_, err = exec.LookPath("rsync")
+	if err != nil {
+		t.Skip("skipping rsync test")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+	t.Cleanup(cancel)
+
+	cliSet, err := k8s.NewKubernetesClientset()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testingNS := "volume-rsync-test-ns-" + rand.String(5)
+
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testingNS,
+		},
+		Spec: corev1.NamespaceSpec{},
+	}
+
+	_, err = cliSet.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		_ = cliSet.CoreV1().Namespaces().Delete(ctx, testingNS, metav1.DeleteOptions{})
+	})
+	t.Log("created namespace: ", testingNS)
+
+	testingPVCName := "testing-pvc"
+
+	err = k8s.CreatePersistentVolumeClaim(ctx, testingPVCName, testingNS,
+		nil, nil,
+		corev1.ReadWriteOnce, *resource.NewQuantity(1024, resource.DecimalSI))
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Log("created PVC: " + testingPVCName)
+
+	l, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() { _ = l.Close() })
+	t.Log("will serve rsync daemon at:", l.Addr())
+
+	go func() {
+		err = k8s.ServeRsyncOnVolume(ctx, l, testingPVCName, testingNS)
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	cmd := exec.Command("rsync",
+		filepath.Join("testData", "content", "hello.txt"),
```

Comment:
```suggestion
		filepath.Join("testdata", "content", "hello.txt"),
```
Since go tool treats the exact name `testdata` as special, this may be best renamed.  From `go help packages`:
> Directory and file names that begin with "." or "_" are ignored
> by the go tool, as are directories named "testdata".
I am pretty sure it's case-sensitive?

Reply:
>I am pretty sure it's case-sensitive

@lkingland  I am not going to argue, but how that would work on non-case sensitive OS?

Reply:
like everything is lower by default?

Reply:
from Go source it looks like it is case-sensitive